docroot/modules/custom/mass_utility/src/Commands/MassUtilityCommands.php

```diff
@@ -398,4 +399,95 @@ public function queueRevisionsCleanup(array $options = [
 
   }
 
+  /**
+   * Copy D2D nodes into redirect entities.
+   *
+   * @command copyd2d
+   * @option limit
+   * @option min-hits
+   */
+  public function copyD2DRedirects($options = ['limit' => self::REQ, 'min-hits' => 50]) {
+    $query = $this->database->select('node__field_legacy_redirects_legacyurl', 'source')
+      ->fields('source', ['entity_id', 'delta']);
+    $query->addField('source', 'field_legacy_redirects_legacyurl_value', 'source');
+    $query->addField('target', 'field_legacy_redirects_ref_conte_target_id', 'target');
+    if ($options['limit']) {
+      $query->range(0, $options['limit']);
+    }
+    $query->innerJoin('node__field_legacy_redirects_ref_conte', 'target', 'source.entity_id = target.entity_id AND source.delta = target.delta');
+    $records = $query->execute()->fetchAll();
+    foreach ($records as $record) {
+      $record->source = parse_url($record->source, PHP_URL_PATH);
```

Comment:
I wonder if we need to deal with query strings?  Not sure

Reply:
Also, we need to drop trailing slashes from the source, if there are any.

Reply:
I didn't see any querystrings in the URLs I looked at.

Drop trailing slashes right before we save as a Redirect entity? Or earlier?

Reply:
I think right here is good, since we'll have direct access to the path (rather than path + query string or whatever).